src/gui/shell.cpp

```diff
@@ -1212,52 +1211,70 @@ void Shell::mouseMoveEvent(QMouseEvent *ev)
 	}
 }
 
+static int normalize(float x)
+{
+    if (x == 0.0f) return 0;
+    else if (x < 0) return -1;
+    else return 1;
+}
+
 void Shell::wheelEvent(QWheelEvent *ev)
 {
 	if (!m_attached || !m_mouseEnabled) {
 		return;
 	}
-#ifdef Q_OS_MAC
-	// For some reason <ScrollWheel*> scrolls multiple lines at once
-	// we have to account for it, to make sure that pixelDelta() is used correctly.
-	const int scroll_columns = 6;
-	const int scroll_rows = 3;
-	// Minimal scroll step
-	int scroll_step_x = cellSize().width() * scroll_columns;
-	int scroll_step_y = cellSize().height() * scroll_rows;
-	// Total scroll delta considering previous events
-	QPoint total_delta = m_mouse_wheel_delta_fraction + ev->pixelDelta();
-	// Delta rounded to a minimal scroll step
-	QPoint cell_delta(total_delta.x() / scroll_step_x, total_delta.y() / scroll_step_y);
-	// Save remainder for future events
-	m_mouse_wheel_delta_fraction = total_delta - QPoint(cell_delta.x() * scroll_step_x, cell_delta.y() * scroll_step_y);
-
-	int horiz = cell_delta.x();
-	int vert = cell_delta.y();
+#if (QT_VERSION < QT_VERSION_CHECK(5, 5, 0))
+	int defaultDeltasPerStep = 120;
 #else
-	int horiz, vert;
-	horiz = ev->angleDelta().x();
-	vert = ev->angleDelta().y();
+	int defaultDeltasPerStep = QWheelEvent::DefaultDeltasPerStep;
+#endif
+
+	QPointF scroll_delta_f(float(ev->angleDelta().x()) / defaultDeltasPerStep,
```

Comment:
Is the float cast here unnecessary?

I think there is an implicit cast based on the constructor's type of `qreal`. Loss of precision also occurs, `qreal` is usually `double`.

Reply:
Was necessary as angleDelta() returns a QPoint but I wrote it now by doing the division after scroll_delta_f was constructed which yields the same result.